### ISSUE TITLE
Implement Federation

### ISSUE THREAD

**Issue by yitsushi** [yitsushi/go-misskey#4]
- [x] federation/followers #30 
 - [x] federation/following #30 
 - [x] federation/instances #42 
 - [x] federation/show-instance #43 
 - [x] federation/users

https://misskey.io/api-doc#tag/federation

**Comment by Skarlso** [comment]
I'll take this :)

**Comment by yitsushi** [comment]
The `ap/show` endpoint will be tricky. It does not have one response. The response is depending on the resource it was requested, the result type will be specified in the `type` field of the root object and the result will be in the `object`.

**Comment by yitsushi** [comment]
I added some response JSON files as a fixture: https://github.com/yitsushi/go-misskey/tree/main/services/federation/fixtures/ap-show

[discussion]
Maybe `ap/show` should live in its own service "client.Ap()" and all possible types will be added as a new function like `client.Ap().ShowNote(string uri)`. It should have only one ShowRequest and a `Show()` function that returns with an `ShowResponse` and all other functions would just call that `Show()` function and typecast the `Object` property and return only with that property.

```
type ShowResponse struct {
  Type string `json:"type"`
  Object interface{} `json:"object"`
}
```

That way if someone wants to query an unimplemented resource or they want to use it as a detector "I got this URI, what is its type?", they can use `Show()` and use the type field to check what its `Type`, but if they know the resource type or they want a specific resource type like Note if it's not a Node they want an error, they can use the `ShowNote()` function.
